### PR TITLE
fix(bench): retry transient provider throttling

### PR DESCRIPTION
**Pull request by joshuaswarren**
## Summary
- extend retryFetch so the configured provider wait budget also covers transient 5xx responses from throttled cloud backends
- parse provider body hints like "try again in N seconds" when retrying 5xx responses
- enforce a minimum 100ms sleep for extended transient retries to avoid tight loops
- preserve default behavior when no extended wait budget is configured
- add regression tests for 429→500→success, minimum 5xx delay, and repeated 5xx without budget

## Validation
- npx tsx --test packages/bench/src/providers/retry-fetch.test.ts packages/bench/src/providers/ollama.test.ts
- npm run preflight:quick (still fails before project checks complete: packages/remnic-core/src/secure-store/kdf.ts cannot resolve @node-rs/argon2)
- npm run review:cursor (unavailable locally: macOS login keychain is locked)

## Benchmark impact
The stopped full AMA run hit Ollama Cloud 429s followed by HTML 500 "try again" responses. Before this change, those became task failures after maxAttempts even though --max-429-wait was set. This makes the run wait through provider-side throttling instead of producing false low scores.

---

> [!NOTE]
> **Medium Risk**
> Changes retry/backoff control flow and timing for provider calls, which could increase wait time or alter failure behavior under certain error patterns; behavior is guarded by `max429WaitMs` and covered by new tests.
> 
> **Overview**
> Improves `retryFetch` so `max429WaitMs` becomes an *extended provider wait budget* that can also keep retrying transient 5xx responses (and transient network errors) beyond `maxAttempts`, preventing benchmark failures during provider-side throttling.
> 
> Adds parsing of body hints like "try again in N seconds" to derive a retry delay for 5xx responses, centralizes remaining-budget calculation, and introduces regression tests for 429→500→success recovery plus ensuring repeated 5xx still fails when no extended budget is configured.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 03998c68ac579b1b230de4abe5ac93f759cb8e63. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/bench/src/providers/retry-fetch.ts [hunk 1]
```diff
@@ -204,19 +216,38 @@ export async function retryFetch(
...
       );
       last429IsStale = true;
+
+      if (attempt >= opts.maxAttempts) {
+        const retryAfter =
+          parseRetryAfterMs(response.headers.get("retry-after")) ??
+          parseBodySuggestedRetryMs(bodyPreview);
+        const waitMs = Math.min(
+          retryAfter ?? opts.baseBackoffMs * Math.pow(2, attempt - 1),
+          remainingExtendedBudgetMs(),
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/bench/src/providers/retry-fetch.ts:241]: Fixed in 03998c68. Extended 5xx retry waits now floor at 100ms before clamping to the remaining provider wait budget, and the regression test asserts the minimum delay for a body hint of "try again in 0 seconds".
